lms/static/scripts/frontend_apps/components/URLPicker.tsx

```diff
@@ -56,13 +56,38 @@ export default function URLPicker({
             : 'Annotating YouTube videos has been disabled at your organisation.',
         );
       } else {
-        onSelectURL(input.current!.value);
+        setError(null);
       }
     } catch (e) {
       setError('Please enter a URL, e.g. "https://www.example.com"');
     }
   };
 
+  // Sync HTML validation state with displayed error.
+  useLayoutEffect(() => {
+    input.current!.setCustomValidity(error ?? '');
+  }, [error]);
```

Comment:
We could do this in `validateInput`, but that would break if ever there was a situation where the form started out in an error state. Doing it this way ensures that the validation state is always in sync with the displayed error.